app/models/work.rb

```diff
@@ -1230,4 +1230,9 @@ def nonfiction
     nonfiction_tags = [125773, 66586, 123921, 747397] # Essays, Nonfiction, Reviews, Reference
     (filter_ids & nonfiction_tags).present?
   end
+
+  # If the creator(s) allow this work to be invited to collections.
+  def allow_collection_invitation
+    users.all? { |user| user == User.current_user || user.preference.allow_collection_invitation }
```

Comment:
As I mentioned above, I think the intention is to not change the behavior of the "Edit Work" form, so it's not an issue if the work's owners can't use the "Add to Collections" form. That said, I think there are a few different ways to approach the behavior of the "Add to Collections" form for work owners:

1. Make it so that the creator of the work can always use the "Add to Collections" form to add their own work to collections, regardless of the number of co-creators or their preferences. This means that the "Add to Collections" form is as permissive as the "Edit Work" page for work owners.
2. Make it so that the creator of the work can only use the "Add to Collections" form if at least one of the work's creators allows collection invitations. I think this approach most closely matches the issue description, but I'm not sure if that's the intended approach.
3. Modify the "Add to Collections" form so that only collection maintainers can use it, and rename it to "Invite to Collections."

My personal preference is for option 3, because I feel like making the form available to both work owners and collection maintainers with the same wording used for both makes it more confusing. But you should probably get someone else's feedback on the desired behavior.

Reply:
I like 3 as well, but I will run it by Support and make sure they don't have any objections.

Reply:
Support is also good with 3!